### PR TITLE
Fix formatted_body being parsed as Markdown

### PR DESCRIPTION
**Pull request by cloudrac3r**
## Type of change

Bugfix

## Content

Background: Clients write Markdown and convert it to HTML before sending the event. All events are formatted as HTML. However, if an HTML formatted event happened to include markdown characters, Element Android would incorrectly render that markdown.

For example, an event with formatted_body: "*test*" should be displayed as literally *test* with no effects, but Element Android incorrectly displayed it as test in italics.

**This commit fixes this behaviour, making Element Android not parse Markdown in HTML messages.**

From the perspective of most users it will appear that backslash escapes now work properly (even though this wasn't the real issue).

## Motivation and context

https://github.com/vector-im/element-android/issues/5657

## Tests

I created 4 messages with the following formatted_body: `*normal*`, `<em>italics</em>`, `* normal`, `<ul><li>list</li></ul>` and I checked that they rendered the same on Android as they do on Desktop. (Desktop is in line with my expectations and does not appear to have any rendering bugs here.)

(I tried various more complex tests, but the ones mentioned above are the important ones that reproduce the behaviour and can be used to test this change.)

It might be good to have automated tests here but I don't know how to do those. If you want to do automated tests then I'll help you by submitting all the messages I used for testing as well as their expected outcomes. Markwon itself has a test suite so we could probably copy the structure of that.

But I'd honestly rather this PR was just merged soonish so that I don't have to be as annoyed while I'm using the app normally. 

## Tested devices

Physical OnePlus 5T, LineageOS 18.1 (Android 11)

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
-  Accessibility has been taken into account - N/A
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- Pull request includes screenshots or videos if containing UI changes - N/A
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- If you have modified the screen flow, or added new screens to the application - N/A

## Notes

Since the changelog.d file contents are most likely to be seen by end users, I put a more natural explanation of the problem in there. The technical explanation about formatted_body and how this problem is really caused is in the commit message. I hope this is all good.

Thanks everybody.

Signed-off-by: Cadence Ember <cloudrac3r@vivaldi.net>

### CODE REVIEW

**Comment by cloudrac3r** [comment]
The CI fail isn't my fault. Can I get a code review, please?

File: vector/src/main/java/im/vector/app/features/html/EventHtmlRenderer.kt [hunk 1]
```diff
@@ -63,14 +68,28 @@ class EventHtmlRenderer @Inject constructor(
...
+        .usePlugin(object : AbstractMarkwonPlugin() {
+            override fun configure(registry: Registry) {
+                registry.require(MarkwonInlineParserPlugin::class.java, { plugin: MarkwonInlineParserPlugin ->
+                    plugin.factoryBuilder().addInlineProcessor(HtmlInlineProcessor())
+                })
+            }
+        })
+        .usePlugin(object : AbstractMarkwonPlugin() {
+            override fun configureParser(builder: Parser.Builder) {
+                builder.enabledBlockTypes(Collections.emptySet())
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/features/html/EventHtmlRenderer.kt:88]: with kotlin the idiomatic way to declare an empty set is with `kotlin.collections emptySet()`

**Review by ouchadam — APPROVED** [review]
some small formatting/replacement comments otherwise LGTM thanks for the fix! :100:

**Comment by cloudrac3r** [comment]
I made the changes you requested and verified on my phone that it still works the same.

If you'd like, I can also add code comments to the parts of the file I touched explaining what the code does. I didn't see any *other* comments in that file, though, so I don't know what your organisation's style is. (Maybe you all rely on git blame instead.)

The CI fails are once again not my fault.

**Comment by ouchadam** [comment]
> If you'd like, I can also add code comments to the parts of the file I touched explaining what the code does. I didn't see any _other_ comments in that file, though, so I don't know what your organisation's style is. (Maybe you all rely on git blame instead.)

this isn't a blocker for me, comments could be handy here as the code is quite decoupled from its usages, in the future a suite of markdown related instrumentation tests would help us avoid regressions in the area and explicitly call out all the formatting cases

**Pull request merged**